games/main.tex

```diff
@@ -0,0 +1,1123 @@
+chequebook contract. For ease of explanation we assume that beneficiary is the
+chequebook contract itself. Channel deposits can then be implemented as a map with beneficiary contract
+address as key and the integer balance in wei as value.
+
+Two chequebook contracts that have a record for each other constitute what is
+called a \gloss{payment channel}.
+
+
+\begin{center}
+\begin{figure}
+\begin{center}
+\begin{tikzpicture}
+  \node (middle)[draw, rectangle, fill=green!60, minimum height=2em, minimum width=8em]{};
+  \node (midleft)[draw, rectangle, fill=orange!60, minimum height=2em, minimum width=4em, node distance=6em, left of=middle]{};
+\node (midright)[draw, rectangle, fill=orange!60, minimum height=2em, minimum width=4em, node distance=6em, right of=middle]{};
+\node (leftred) [draw, rectangle, fill=red!60, minimum height=2em, minimum width=2em, node distance=9em,left of=middle]{};
+\node (rightred)[draw, rectangle, fill=red!60, minimum height=2em, minimum width=2em, node distance=9em,right of=middle]{};
+\node (zero) [above of=middle,node distance=3.5em, text width=4em, align=center] {\small zero\\ balance};
+\node (zerod) [below of=middle] {};
+\draw [dashed](zero)--(zerod);
+\node (rtol) [node distance=8em,right of=zero,text width=4em, align=center] {\small payment\\threshold};
+\node (rtold) [node distance=8em,right of=zerod] {};
+\node (ltol) [node distance=8em,left of=zero,text width=4em, align=center] {\small payment\\threshold};
+\node (ltold) [node distance=8em,left of=zerod] {};
+\node (prtol) [node distance=4em,right of=zero,text width=4em, align=center] {\small channel\\deposit};
+\node (prtold) [node distance=4em,right of=zerod] {};
+\node (pltol) [node distance=4em,left of=zero,text width=4em, align=center] {\small channel\\deposit};
+\node (pltold) [node distance=4em,left of=zerod] {};
+\draw [dashed](rtol)--(rtold);
+\draw [dashed](ltol)--(ltold);
+\draw [dashed](prtol)--(prtold);
+\draw [dashed](pltol)--(pltold);
+\end{tikzpicture}
+\end{center}
+\caption{}
+\label{fig:paymentchannel}
+\end{figure}
+\end{center}
+
+\subsection{Zero cost entry}
+
+An additional advantage of this construct is that one can start providing service with zero ether.
+Since swap can also work only one way, if a party enters the system with zero ether (newcomer),
+but connects to peers with funds (insider), first they just provide the service (and not use it)
+in order to produce a positive balance. In the simplest case insiders pay the newcomer on chain
+but there are more efficient ways of bootstrapping.
+
+If a newcomer connects to  an insider with an existing chequebook contract,
+the insider agrees to create a chequebook contract for the newcomer
+in return for consuming newcomers services.
+ When insider's service debt reaches the amount needed by for contract creation,
+ insider just sends a transaction to create the contract with newcomer as the owner.
```

Comment:
So the insider pay the newcomer by setting a checkbook contract for him as an owner ?
nice!